### ISSUE TITLE
[feature request] Improved remotes with multiple (re-usable)  customizable IR data sets (devices), activities, macros & entities

### ISSUE THREAD

**Issue by F4Ck0R** [unfoldedcircle/bug-reports#112]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Based on my request: https://github.com/unfoldedcircle/feature-and-bug-tracker/issues/110 It would also be great to be able to create device IR Code sets. For example I'd create a Device like my Denon X4500H with all IR Codes that the AVR supports and create a Device for my LG 77C2 with all the IR Codes that my TV supports. Afterwards I create a Remote (entity to be selected to show on remote UI) and select the different buttons I want to use in this remote from the devices I just created.

Maybe I need another remote with only my Denon X4500H for controlling music with only Spotify Connect without TV being involved but other Buttons / Components on the remote.

By creating devices first, I don't need to setup everything individually again. I could just re-use whats already there. If I remember correctly this is also how Logitech Harmony is setup. Select all the **devices** you want to control and afterwards use them in remotes, activities, etc.

Additionally, I might also want control an entity such as turning on/off a light or execute a macro or activity from a selected remote. Without going back and forth.

You could do so by letting users:
- safe and re-use their (custom) IR data set as a "device to be controlled"
- select multiple "devices" (IR data sets) in one remote, activity, macro etc.
- choose the buttons / components as well as entities (certain function or a link to the entity page), macros and activities to put them to the remote page or assign them to a phyisical button of the RemoteTwo

I hope my description was understandable.

Kind regards

**Comment by F4Ck0R** [comment]
BTW: This is going into a similar direction: https://github.com/unfoldedcircle/feature-and-bug-tracker/issues/4
Didn't found it in first place since it is tagged as a bug. On this page but also afterwards selection of multiple IR data sets (devices) should be possible.

**Comment by r00li** [comment]
Yep, I also really want this. I created an IR remote for my blu-ray player, but I want to control the volume via my AV receiver. However I don't want to create a new activity with these two devices and configure the remote layout again. I just want to add the volume up/down commands from my other remote to this one.

**Comment by zehnm** [comment]
> I created an IR remote for my blu-ray player, but I want to control the volume via my AV receiver.

That's where activities come into play:
- 2 remote-entities: Blu-ray player & AVR
- Activity with the 2 remote entities

In the activity you can mix-and-match commands from all included entities. Control the player and map volume buttons to the AVR.
A remote-entity is just to control **one** IR device.

As a workaround you can also define custom buttons in a manufacturer IR dataset of an IR remote, or assign other IR codes to existing buttons.
Example: if you just want a remote entity with AVR volume control but without creating an activity:

1. Create a new remote for the Blue-ray player. Either a manufacturer or custom device.
2. Edit the IR dataset and copy / paste the PRONTO code of your AVR functions (or learn the codes).

<img width="702" alt="image" src="https://github.com/unfoldedcircle/feature-and-bug-tracker/assets/17356629/12832d69-9f58-4f67-a714-c5e4963ea8fc">

- The `Generic Blu-ray 1` IR remote is from a manufacturer device.
- It doesn't contain `VOLUME` buttons
- One can add custom buttons: define VOLUME_UP & VOLUME_DOWN

If the common IR button names are used, the auto-mapping to physical keys & UI components _should_ work.

A support article will follow, how IR buttons should be named, to get the auto-mapping features and future enhancements based on well-known button names. These are the most important ones:
```
BACK
CURSOR_DOWN
CURSOR_LEFT
CURSOR_RIGHT
CURSOR_UP
CURSOR_ENTER
DIGIT_0
DIGIT_1
DIGIT_2
DIGIT_3
DIGIT_4
DIGIT_5
DIGIT_6
DIGIT_7
DIGIT_8
DIGIT_9
HOME
MENU
MUTE_TOGGLE
POWER_OFF
POWER_ON
POWER_TOGGLE
VOLUME_DOWN
VOLUME_UP
```

**Comment by r00li** [comment]
The workaround of learning the AV receiver codes again and adding them to my blu-ray player is what I did in the end. But it is not ideal. I would go with activity, but I already had everything set up and then I would need to prepare the whole layout again just for two buttons. So it would be great if we could at least start creating the activity by using the layout from an existing remote as a starting point.
